dsa-23au/java-dsa/jpract-app/src/main/java/com/practice/app/mergeSort.java

```diff
@@ -0,0 +1,75 @@
+package main.java.com.practice.app;
+
+public class mergeSort 
+{
+    public void sort(int[] inputArray)
+    {
+		
+		//Instantiating the variables length and midpoint
+		int length = inputArray.length;
+		int midpoint = length / 2;
+		
+		//setting leftSide size equal to the length
+		//up to midpoint
+		//setting rightSide equal to the remaining length
+		int[] leftSide = new int[midpoint];
+		int[] rightSide = new int[length - midpoint];
+		
+		//if the length is equal to 2, it is already
+		//in order, there is no dividing possible
+		//or necessary
+		if (length < 2)
+		{
+			return;
+		}
+		
+		//for inputs greater than that it is splitting them
+		//by filling the leftSide and rightSide
+		else
+		{
+			for(int i = 0; i < midpoint; i++)
+			{
+				leftSide[i] = inputArray[i];
+			}
+			for(int j = midpoint; j < length; j++)
+			{
+				rightSide[j - midpoint] = inputArray[j];
+			}
```

Comment:
Since j = midpoint, then on line 36 the j - midpoint will = 0 repeating line 32. This will just keep creating new arrays .